slidey_js.js

```diff
@@ -0,0 +1,101 @@
+$( document ).ready(function() {
+    console.log( "ready!" );
+});
+  var background = $("#background")
+  enMarche = 0
+
+  //
+  // var getAPI = function() {
+  //   var giphyAPI = "http://api.giphy.com/v1/gifs/search?q=ryan+gosling&api_key=dc6zaTOxFJmzC&limit=5";
+  //   $.getJSON( giphyAPI, {
+  //     // tags: "mount rainier",
+  //     // tagmode: "any",
+  //     format: "json"
+  //   })
+  // }
+  //
+  //
+  // var xhr = $.getJSON("http://api.giphy.com/v1/gifs/search?q=ryan+gosling&api_key=dc6zaTOxFJmzC&limit=5");
+  // console.log(xhr);
+  // xhr.done(function(data) { console.log("success got data", data); });
+
+  var imageArray = ["http://unidevphp.com/php-projects/penymobile/images/auction_images/7_564901_images.jpg", "http://i.dailymail.co.uk/i/pix/2015/01/06/2473100D00000578-2898639-Photographer_Andrey_Gudkov_snapped_the_images_on_the_plains_of_t-a-20_1420546215677.jpg", "http://www.zastavki.com/pictures/originals/2013/Photoshop_Image_of_the_horse_053857_.jpg", "http://www.hdwallpapersimages.com/wp-content/uploads/2014/02/Magic-Of-Blue-Universe-Images.jpg"]
+
+  var counter = 0;
+  var changeBackground = function(){
+    if (enMarche = 1){
+      $("#background").css("background-image", "url('" + imageArray[counter] + "')")
+      console.log(imageArray[counter])
+      counter = counter + 1
+      if (counter >= imageArray.length){
+        counter = 0;
+      }
+    }
+  }
+
+  var startImageSwap = function(){
+    enMarche = 1
+    changeBackground();
+    console.log("bob")
+    interval = setInterval(changeBackground, 1000);
+  };
+
+  startImageSwap();
+
+$(window).on("click", function(){
+  changeBackground();
+  interval = setInterval(changeBackground, 1000);
+  });
+  //
+  // pause.addEventListener("click", function(){
+  //   clearInterval(interval);
+  //   enMarche = 0
+  // })
+
```

Comment:
So you have some code commented out here, particularly it seems your trying to nest a click event within the current click event. My thought instead would be to just instantiate an additional variable in the scope of this function. The use case for this variable will be to just know whether the `setInterval` is in progress or not. maybe something like: `var slideShowOccuring = true` That way when it's true you can change the nature of the click event to `clearInterval(<setIntervalVarGoesHere>)` instead of setting the interval. You could nest all of this logic in an if/else conditional.